File: 0440-20-2780-minimum-index-of-a-valid-split/0440-20-2780-minimum-index-of-a-valid-split.py
class Solution:
    def minimumIndex(self, nums: List[int]) -> int:
        cnt2 = 0
        for num in nums:
            if not cnt2:
                x = num
            cnt2 += (num == x)
            cnt2 -= (num != x)
        
        cnt2 = nums.count(x)
        n = len(nums)
        cnt1 = 0
        
        for i in range(n):
            cnt1 += (nums[i] == x)
            cnt2 -= (nums[i] == x)
            if cnt1 * 2 > i + 1 and cnt2 * 2 > n - (i + 1):
                return i
        
        return -1

File: 0440-20-2780-minimum-index-of-a-valid-split/0440-20-2780-minimum-index-of-a-valid-split.java
class Solution {
    public int minimumIndex(List<Integer> nums) {
        int cnt2 = 0, x = 0;
        for (int num : nums) {
            if (cnt2 == 0) x = num;
            cnt2 += (num == x) ? 1 : -1;
        }
        
        cnt2 = 0;
        for (int num : nums) {
            if (num == x) cnt2++;
        }
        
        int n = nums.size();
        int cnt1 = 0;
        
        for (int i = 0; i < n; i++) {
            cnt1 += (nums.get(i)  == x) ? 1 : 0;
            cnt2 -= (nums.get(i)  == x) ? 1 : 0;
            
            if (cnt1 * 2 > i + 1 && cnt2 * 2 > n - (i + 1)) {
                return i;
            }
        }
        
        return -1;

    }
}